[doc] update CONTRIBUTING.md (#473)
Schiano-NOAA/ASAR: CONTRIBUTING.md

### Recommended workflow: fork & submit a pull request

The most efficient way to contribute an idea is to fork `asar`, make your 
suggested changes on a local branch, and then submit a pull request to the main branch. 
This will allow the developers to easily evaluate your suggested changes. 
Please see the [GitHub Docs' "Contributing to a project" page](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project) 
for step-by-step guidance in using this workflow.

We also follow a few other practices that will help us expedite the review process.
After completing the bug fix or feature, please complete the following:

1. Run `devtools::test()` to verify the package checks are passing (this includes tests).

2. If including a feature, please add a test using the 
[`{testthat}` package structure](https://testthat.r-lib.org/reference/test_that.html).
If you are unfamiliar with this approach, simply let us know :smile

3. If including a feature that is a new function, please add documentation using
the [`{roxygen2}` package structure](https://roxygen2.r-lib.org/articles/roxygen2.html).

> 4. Please increment the package version following our [semantic versioning section](https://github.com/nmfs-ost/asar/blob/doc-update-contrib/CONTRIBUTING.md#semantic-versioning-sec-version).

Reviewer: Is this section present? I don't see it

Author: Oops no, not anymore. I removed it after re-thingking about including our semantic versioning rules as it wouldn't be relevant to a contributor.

> -   Ensure your base branch is correct. We typically merge most PRs into "dev", then merge "dev" into "main" upon a new release.

Reviewer: ```suggestion
-   Ensure your base branch is correct. We merge all branches into main.
```

#### Release Indicators:

| Indicator | Description |
|-----------|-------------|
| feat | A new feature |
| fix | A bug fix |
| docs | Documentation only changes |
| style | Changes that do not affect the meaning of the code (i.e. white-space, formatting, missing semi-colons) |
| refactor | A code change that neither fixes a bug nor adds a feature |
| perf | A code change that improves performance |
| test | Adding missing or correcting existing tests |
| chore | changes to the build process or auxiliary tools and libraries such as documentation generation |

#### Rules for Indicator Use

- If the body contains the text "BREAKING CHANGE" then MAJOR version is incremented.
- If the type contains feat, then MINOR version is incremented.
- If the type contains a fix, then PATCH version is incremented.
- If the type contains refactor/style/perf/doc/test/chore, then nothing is incremented and no release will be made.

**The above content is modeled after an article on ["Automating Versioning and Releases Using Semantic Release"](https://medium.com/agoda-engineering/automating-versioning-and-releases-using-semantic-release-6ed355ede742)

> # from Agoda Engineering.**

Reviewer: ```suggestion
**The above content is modeled after an article on ["Automating Versioning and Releases Using Semantic Release"](https://medium.com/agoda-engineering/automating-versioning-and-releases-using-semantic-release-6ed355ede742) from Agoda Engineering.**
```
